src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -604,7 +604,8 @@ BinaryExpressionSyntax or
                 AssignmentExpressionSyntax or
                 QueryClauseSyntax or
                 SelectOrGroupClauseSyntax or
-                CheckedExpressionSyntax)
+                CheckedExpressionSyntax or
+                WhenClauseSyntax)
```

Comment:
this should move into the 'Kind-check' switch above (as there's only one kind of WhenClause.  Otherwise, this is all good :)